src/elm/Pages/Deployments/View.elm

```diff
@@ -64,56 +66,63 @@ addForm deploymentModel =
 
 {-| viewPreview : renders single deployment item preview
 -}
-viewPreview : Org -> Repo -> Deployment -> Html msg
-viewPreview org repo deployment =
+viewPreview : Posix -> Zone -> Org -> Repo -> Deployment -> Html msg
+viewPreview now zone org repo deployment =
     let
         deploymentId =
             String.fromInt deployment.id
 
-        status =
-            [ buildStatusToIcon Vela.Success ]
-
-        commit =
-            [ text <| deployment.target ++ " (" ++ Util.trimCommitHash deployment.commit ++ ")" ]
-
-        ref =
-            [ text deployment.ref ]
-
-        sender =
-            [ text deployment.user ]
+        info =
+            div [ class "deployment-info" ]
+                [ div []
+                    [ p [] [ text ("#" ++ deploymentId) ]
+                    , p [] [ text deployment.task ]
+                    ]
+                , div []
+                    [ p [] [ text (deployment.target ++ " at (" ++ Util.trimCommitHash deployment.commit ++ ")") ]
+                    , p [] [ text (" Deployed by " ++ deployment.user) ]
+                    ]
+                ]
 
-        description =
-            [ text ("- " ++ deployment.description) ]
+        promoteDeploymentLink =
+            div [ class "deployment-link" ] [ a [ Routes.href <| Routes.PromoteDeployment org repo deploymentId ] [ text "Deploy" ] ]
 
-        id =
-            [ text ("#" ++ deploymentId) ]
+        deploymentDetails =
+            div [ class "deployment-details" ]
+                [ p [] [ text (" Task: " ++ deployment.task) ]
+                , p [] [ text (" Ref: " ++ deployment.ref) ]
+                , p [] [ text <| " Description: " ++ deployment.description ]
+                ]
 
-        redeploy =
-            [ a [ Routes.href <| Routes.PromoteDeployment org repo deploymentId ] [ text "Redeploy" ] ]
+        builds =
+            viewDeploymentBuilds now zone org repo <| deployment.builds
 
-        markdown =
-            [ div [ class "status", Util.testAttribute "deployment-status", class "-success" ]
-                status
-            , div [ class "info" ]
-                [ div [ class "row -left" ]
-                    [ div [ class "id" ] id
-                    , div [ class "commit-msg" ] [ strong [] description ]
+        status =
```

Comment:
I would argue this functionality and anything that produces Svg elements belongs in the `SvgBuilder` module.